gax-grpc/src/test/java/com/google/api/gax/grpc/CancellationHelpers.java

```diff
@@ -47,7 +47,9 @@ public static void cancelInThreadAfterLatchCountDown(
               public void run() {
                 try {
                   latch.await();
-                  while (!resultFuture.cancel(true)) {
+                  System.out.println("Cancelling");
+                  while (!resultFuture.cancel(true) && !resultFuture.isDone()) {
+                    System.out.println("Cancelling again");
```

Comment:
Removed the statements. 
As for why cancelling again: any future (represented by a specific future task, for example `com.google.common.util.concurrent.ListenableFutureTask`, which is the most common in our code), when cancel() is called may or may return false, meaning that cancellation failed. 

Documentation says that most probably it is because the future is already completed/cancelled, but `com.google.common.util.concurrent.ListenableFutureTask` sometimes return false even for the still running futures. I believe (but I is not necessarily the case) this happens because of the rare condidtion, when same FutureTask is being set and cancelled at the same time from different threads, and since Future.set() and Future.cancel() use non-blocking algorithms, if race condition happen they just return `false` instead of blocking and waiting for the thing to proceed "properly".

This becomes even more common case if we are running the AttemptCallable, which explicitly blocks cancellation for a short amount of time setting retryingFuture attempt future to NonCancellableFuture.